trilium_alchemy/core/note/note.py

```diff
@@ -962,14 +952,46 @@ def __getitem__(self, key: str | int) -> str | Note:
             else:
                 return attr.value
 
-    def __setitem__(self, key: str | int, value_spec: ValueSpec):
+    def __setitem__(self, key: str, value_spec: ValueSpec):
         """
-        Create or update attribute with provided name.
+        Create or update attribute with provided name or index.
+
+        :param key: Attribute name
+        :param value_spec: Attribute value
         """
         self.attributes[key] = value_spec
 
-    def __contains__(self, key: str) -> bool:
-        return key in self.attributes._name_map
+    def __hash__(self):

```

Comment:
Was there a reason this was needed? `self.note_id` may be `None` so it can't be used this way. The default mechanism of using `id()` as the hash value should work fine.

I actually had it this way originally but ran into problems with newly created notes having hash collisions, then realized it works fine without implementing `__hash__` at all.

Reply:
I had to add __hash__ back in as there were notes used as dictionary keys. I will try id() to see how that works.

Reply:
Ah I see. `Entity` objects are added to a set when flushing as well, so in general they need to be hashable. Python was automatically using `id()` before we subclassed `MutableMapping` as I understand, so this looks like the right thing.